Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1017,57 +1017,55 @@ namespace OpenInfraPlatform {
 
-							double radiusMax = std::max(xRadius, yRadius);
-							double radiusMin = std::min(xRadius, yRadius);
-							int numSegments = GeomSettings()->getNumberOfSegmentsForTessellation(radiusMin);
-							double deltaAngle = GeomSettings()->getAngleLength(radiusMin);
+					// Calculate a number of segments.
+					int numSegments = GeomSettings()->getNumberOfSegmentsForTessellation(radiusMax);
+					double deltaAngle = 2.0 * M_PI / numSegments;
 
-							// todo: implement clipping
-							if (!trim1Vec.empty() || !trim2Vec.empty()) {
-								//Calculate an angle on the ellipse (with ellipse center in (0., 0., 0.)) for trimming begin.
-								carve::geom::vector<3> point = getPointOnCurve<typename IfcEntityTypesT::IfcEllipse>(ellipse, trim1Vec, trimmingPreference);
-								double startAngle = getAngleOnEllipse(carve::geom::VECTOR(0, 0, 0), xRadius, yRadius, point);
+					if (!trim1Vec.empty() || !trim2Vec.empty()) {
+						//Calculate an angle on the ellipse (with ellipse center in (0., 0., 0.)) for trimming begin.
+						carve::geom::vector<3> point = getPointOnCurve<typename IfcEntityTypesT::IfcEllipse>(ellipse, trim1Vec, trimmingPreference);
+						double startAngle = getAngleOnEllipse(carve::geom::VECTOR(0, 0, 0), xRadius, yRadius, point);
 
-								//Calculate an angle on the ellipse (with ellipse center in (0., 0., 0.)) for trimming end.
-								point = getPointOnCurve<typename IfcEntityTypesT::IfcEllipse>(ellipse, trim2Vec, trimmingPreference);
-								double endAngle = getAngleOnEllipse(carve::geom::VECTOR(0, 0, 0), xRadius, yRadius, point);
+						//Calculate an angle on the ellipse (with ellipse center in (0., 0., 0.)) for trimming end.
+						point = getPointOnCurve<typename IfcEntityTypesT::IfcEllipse>(ellipse, trim2Vec, trimmingPreference);
+						double endAngle = getAngleOnEllipse(carve::geom::VECTOR(0, 0, 0), xRadius, yRadius, point);
 
-								// Calculate an opening angle.
-								double openingAngle = calculateOpeningAngle(senseAgreement, startAngle, endAngle);
+						// Calculate an opening angle.
+						double openingAngle = calculateOpeningAngle(senseAgreement, startAngle, endAngle);
 
-								numSegments = GeomSettings()->getNumberOfSegmentsForTessellation(radiusMin, abs(openingAngle));
-								deltaAngle = openingAngle/ numSegments;
-							}
+						// Calculate a number of segments.
+						numSegments = GeomSettings()->getNumberOfSegmentsForTessellation(radiusMax, abs(openingAngle));
+						deltaAngle = openingAngle/ numSegments;
+					}
 
-							std::vector<carve::geom::vector<3> > ellipsePoints;
-							double angle = 0.0;
-							for (int i = 0; i < numSegments; ++i) {
-								ellipsePoints.push_back(carve::geom::vector<3>(
-									carve::geom::VECTOR(
-										xRadius * cos(angle),
-										yRadius * sin(angle),
-										0)));
-								angle += deltaAngle;
-							}
+					std::vector<carve::geom::vector<3> > ellipsePoints;
+					double angle = 0.0;

```

Comment:
Here, it would make sense to use `startAngle` - otherwise all your ellipses will always be drawn from 0 to openingAngle and not from startAngle to endAngle. Please check for IfcCircle as well.

Reply:
Resolved: 39fe56e